build-keyring

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -euo pipefail
+
+GPG=${GPG:-gpg}
+
+if [ $# -lt 2 ]; then
+    echo "Usage: $0 KEYRING KEY..." >&2
+    exit 1
+fi
+
+KEYRING=$1
+shift
+
+# Setup temporary GPG homedir
+gpghome=$(mktemp -d gpghome-XXXXXXXX)
+trap "rm -rf \"${gpghome}\"" EXIT
```

Comment:
Shellcheck emitted one warning for this file, pointing at this line:

```
In build-keyring line 17:
trap "rm -rf \"${gpghome}\"" EXIT
               ^--------^ SC2064: Use single quotes, otherwise this expands now rather than when signalled.

For more information:
  https://www.shellcheck.net/wiki/SC2064 -- Use single quotes, otherwise this...
```

Here I think we don't care. You could put:

```suggestion
# shellcheck disable=2064
trap "rm -rf \"${gpghome}\"" EXIT
```

if you cared to.

Reply:
Right, I forgot all `gpg-agent`. I recently became [very](https://github.com/ostreedev/ostree/commit/4670837f7c9d027b8724c4061219ee951dc8cd6e) [familiar](https://github.com/ostreedev/ostree/commit/d14472a7f04d02d392d2754cfb3cec881aadc704) with this feature. It's available reliably since 2.1.17.

eos3.3 has 2.1.18, so we can probably rely on that working. However, the more robust thing to do is kill it before wiping the directory.

For the shellcheck thing, I was intentionally trying to expand the variable immediately, so I think disabling the check is appropriate. However, since I'll probably expand the cleanup into a function to also kill the agent, I can do it a little more appropriately there.

Reply:
Joy! Up to you whether you want to revise this.